pkg/health/health.go

```diff
@@ -68,8 +76,16 @@ func (h ConsulHealthChecker) WatchNodeService(nodename string, serviceID string,
 					catalogResults = append(catalogResults, outResult)
 				}
 				kvCheckResult := consulWatchToResult(kvCheck)
-				catalogCheckResult := findWorstResult(catalogResults)
-				resultCh <- findBestResult([]Result{kvCheckResult, catalogCheckResult})
+				catalogCheckResult, HEErr := findWorstResult(catalogResults)
+				if HEErr != nil {
+					errCh <- HEErr
```

Comment:
In a world where we exclusively do KV checks, this will be an unending stream of errors. The outside world should only care about the case where _both_ error checks are blank.